lib/msal-core/src/UserAgentApplication.ts

```diff
@@ -411,23 +442,29 @@ export class UserAgentApplication {
                         this.logger.error("Error occurred during unified cache ATS: " + error);
 
                         // proceed to login since ATS failed
-                        this.acquireTokenHelper(null, interactionType, isLoginCall, request, scopes, resolve, reject);
+                        this.acquireTokenHelper(null, interactionType, isLoginCall, request, scopes, resolve, reject, state);
                     });
                 }
                 // No ADAL token found, proceed to login
                 else {
-                    this.acquireTokenHelper(null, interactionType, isLoginCall, request, scopes, resolve, reject);
+                    this.acquireTokenHelper(null, interactionType, isLoginCall, request, scopes, resolve, reject, state);
                 }
             }
             // AcquireToken call, but no account or context given, so throw error
             else {
                 this.logger.info("User login is required");
-                throw ClientAuthError.createUserLoginRequiredError();
+                const stateOnlyResponse = buildResponseStateOnly(this.getAccountState(request && request.state));
```

Comment:
we use request && request.state here but not the state passed in?

Reply:
Edited this. Thanks, good catch.